### DISCUSSION TITLE
GitHub Rosie doesn't recognise Ligand chain

### DISCUSSION THREAD

**Discussion by amitmhbh** [RosettaCommons/rosetta#97] [ROSIE]
Hi all,
I prepared a docking model for RossetaDock 5, to be calculated through gitHub.
The protein has 3 chains (A,B,C). My ligand (chain D) is marked UNL, and all have hydrogens added. 
The problem is, that when the pdb is uploaded, the server doesn't recognise the ligand. (it only shows chains ABC)
So far following recommendations online, I tried, unsuccessfully:
1. Save the PDB either through chimeraX and Pymol, in both the ligand is visible. Only when uploaded to RosettaDock, it isn't.
2. Fiddle with the "TER" 
3. Number each atom of the ligand (initially they were all "C", "O" etc, now they are "C1", "C2" and so on).

Does RD5 doesn't like UNL? if so, what should it be instead?
Any other suggestions?

Many thanks!

**Comment by roccomoretti** [comment]
Rosetta has specific checks to discard UNL residues -- UNL is the "unknown ligand" code of the PDB, and it's explicitly reserved for ligands without any consistent (or even known) chemical identity or connectivity. There not really a way to model such residues consistently.

Pretty much any other three letter code (aside from the standard ones) should work. LG1, LG2, etc. or X01, X02, etc. are common choices for arbitrary ligands, but most others should work (even if the wwPDB has it assigned to a particular chemical identity, that can be overridden)

I'll note that the protein-protein docking code in ROSIE (the standard RosettaDock-5.0) isn't set up for working with arbitrary ligand residues. It's mainly intended for protein-protein docking only. The underlying Rosetta protocol should be able to handle the presence of ligands in the complex, but it may only understand the chemical layout of a certain subset of pre-defined ligands. In order to handle arbitrary chemical entities you need to prepare and provide a Rosetta residue parameter file, which would likely require using the command-line version of Rosetta.

I'll also note that using ligands with RosettaDock-5.0 is only recommended if it's primarily a protein-protein interaction. That is, if the goal is to dock the AB complex with the C chain and the C chain just happens to contain the D ligand. If you're interested in docking the D small molecule ligand with the ABC protein complex, then you're looking more at the [Ligand Docking](https://rosie.rosettacommons.org/ligand_docking) protocol, which has explicit steps to set up the parameter file from an arbitrary ligand description (i.e. as an SDF specification.)

**Comment by amitmhbh** [comment]
Hi Rocco,
Thank you so much for the detailed explanation.
Yes, I know that UNL is just the generic code for any ligand, so thank you for suggesting alternative names.
I've submitted a few jobs in parallel to the rosie server, but they take a substantial amount of time to start, and I was hoping for something a bit more powerful.

Thanks again for your suggestions and help,
Cheers

**Comment by roccomoretti** [comment]
Marking this as "closed", as I hope your issue has been adequately addressed. If that assessment is incorrect, feel free to re-open or open up a new discussion with more details.
